### ISSUE TITLE
Can't set password with space using setpwd.sh, setpwd.sh fails

### ISSUE THREAD

**Issue by EgonFreeman** [roubles/macmounter#2]
Can't use setpwd.sh:

```
studio: *$ /DocTor/macmounter/sh/setpwd.sh
Account:nas01_mount
Service:nas01_mount
Password:Got pwd: *****************
Usage: add-generic-password [-a account] [-s service] [-w password] [options...] [-A|-T appPath] [keychain]
    -a  Specify account name (required)
    -c  Specify item creator (optional four-character code)
    -C  Specify item type (optional four-character code)
    -D  Specify kind (default is "application password")
    -G  Specify generic attribute (optional)
    -j  Specify comment string (optional)
    -l  Specify label (if omitted, service name is used as default label)
    -s  Specify service name (required)
    -p  Specify password to be added (legacy option, equivalent to -w)
    -w  Specify password to be added
    -A  Allow any application to access this item without warning (insecure, not recommended!)
    -T  Specify an application which may access this item (multiple -T options are allowed)
    -U  Update item if it already exists (if omitted, the item cannot already exist)

By default, the application which creates an item is trusted to access its data without warning.
You can remove this default access by explicitly specifying an empty app pathname: -T ""
If no keychain is specified, the password is added to the default keychain.
        Add a generic password item.
studio: *$
```

System version: OS X 10.9.5

Trying to get the password, predictably, results in

```
studio: *$ /DocTor/macmounter/sh/getpwd.sh nas01_mount
security: SecKeychainSearchCopyNext: The specified item could not be found in the keychain.
studio: *$  
```

**Comment by EgonFreeman** [comment]
Correction. I filled it out the same as login, and I still get the usage listing, which I assume is a bad thing.

 ```
studio: *$ /DocTor/macmounter/sh/setpwd.sh
Account:nas01_mount
Service:nas01_mount
Password:Got pwd: *****************
Usage: add-generic-password [-a account] [-s service] [-w password] [options...] [-A|-T appPath] [keychain]
    -a  Specify account name (required)
    -c  Specify item creator (optional four-character code)
    -C  Specify item type (optional four-character code)
    -D  Specify kind (default is "application password")
    -G  Specify generic attribute (optional)
    -j  Specify comment string (optional)
    -l  Specify label (if omitted, service name is used as default label)
    -s  Specify service name (required)
    -p  Specify password to be added (legacy option, equivalent to -w)
    -w  Specify password to be added
    -A  Allow any application to access this item without warning (insecure, not recommended!)
    -T  Specify an application which may access this item (multiple -T options are allowed)
    -U  Update item if it already exists (if omitted, the item cannot already exist)
 
By default, the application which creates an item is trusted to access its data without warning.
You can remove this default access by explicitly specifying an empty app pathname: -T ""
If no keychain is specified, the password is added to the default keychain.
        Add a generic password item.
studio: *$
```

OS X 10.9.5

**Comment by roubles** [comment]
I just tried this on OSX 10.11.3.

```
$ ./setpwd.sh
Account:blah
Service:blah
Password:Got pwd: blah
$ ./getpwd.sh blah
blah
```

From your logs, it seems like the set password is not taking. Can you please run the following command on your system and share the output:

```
security add-generic-password -a blah -s blah -w blah
```

**Comment by EgonFreeman** [comment]
**NOTE: fix for the issue present in the .sh script - below.**

```
studio:~ *$ security add-generic-password -a blah -s blah -w blah
studio:~ *$ 
```

and then

```
studio:~ *$ /DocTor/macmounter/sh/getpwd.sh blah
blahstudio:~ *$ 
```

(NOTE: I have retained the lack of a following newline, all completely understandable)

I have rebooted the Mac since the last attempts. With the above being successful, I retried the setpwd.sh script and it ran fine this time. Digging a little deeper, it seems to be an issue with password characters (or characters in general). Consider:

```
studio:~ *$ /DocTor/macmounter/sh/setpwd.sh
Account:test
Service:test
Password:Got pwd: super test
studio:~ *$
```

vs

```
studio:~ *$ /DocTor/macmounter/sh/setpwd.sh
Account:test2 
Service:test2
Password:Got pwd: A lengthy passqword with spaces and numbers 45.
Usage: add-generic-password [-a account] [-s service] [-w password] [options...] [-A|-T appPath] [keychain]
    -a  Specify account name (required)
    -c  Specify item creator (optional four-character code)
    -C  Specify item type (optional four-character code)
    -D  Specify kind (default is "application password")
    -G  Specify generic attribute (optional)
    -j  Specify comment string (optional)
    -l  Specify label (if omitted, service name is used as default label)
    -s  Specify service name (required)
    -p  Specify password to be added (legacy option, equivalent to -w)
    -w  Specify password to be added
    -A  Allow any application to access this item without warning (insecure, not recommended!)
    -T  Specify an application which may access this item (multiple -T options are allowed)
    -U  Update item if it already exists (if omitted, the item cannot already exist)

By default, the application which creates an item is trusted to access its data without warning.
You can remove this default access by explicitly specifying an empty app pathname: -T ""
If no keychain is specified, the password is added to the default keychain.
        Add a generic password item.
studio:~ *$ 
```

The problem is with the space character, it seems:

```
studio:~ *$ /DocTor/macmounter/sh/setpwd.sh
Account:test4
Service:test4
Password:Got pwd: Test 123.
studio:~ *$ /DocTor/macmounter/sh/setpwd.sh
Account:test5
Service:test5
Password:Got pwd: Test 12_3.
studio:~ *$ 
```

but then

```
studio:~ *$ /DocTor/macmounter/sh/setpwd.sh
Account:test3
Service:test3
Password:Got pwd: Test 12 3.
Usage: add-generic-password [-a account] [-s service] [-w password] [options...] [-A|-T appPath] [keychain]
    -a  Specify account name (required)
    -c  Specify item creator (optional four-character code)
    -C  Specify item type (optional four-character code)
    -D  Specify kind (default is "application password")
    -G  Specify generic attribute (optional)
    -j  Specify comment string (optional)
    -l  Specify label (if omitted, service name is used as default label)
    -s  Specify service name (required)
    -p  Specify password to be added (legacy option, equivalent to -w)
    -w  Specify password to be added
    -A  Allow any application to access this item without warning (insecure, not recommended!)
    -T  Specify an application which may access this item (multiple -T options are allowed)
    -U  Update item if it already exists (if omitted, the item cannot already exist)

By default, the application which creates an item is trusted to access its data without warning.
You can remove this default access by explicitly specifying an empty app pathname: -T ""
If no keychain is specified, the password is added to the default keychain.
        Add a generic password item.
studio:~ *$
```

So, the problem seems to be the second space. On closer inspection of the script, I propose a **very** simple fix:

find

```
security add-generic-password -a ${account} -s ${service} -w ${password}
```

replace with

```
security add-generic-password -a "${account}" -s "${service}" -w "${password}"
```

Result:

```
studio:~ *$ /DocTor/macmounter/sh/setpwd.sh
Account:test7
Service:test7
Password:Got pwd: This Is a Very Long Pasword with some nice entropy, blah blah 123 456 789.
studio:~ *$ /DocTor/macmounter/sh/getpwd.sh test7
This Is a Very Long Pasword with some nice entropy, blah blah 123 456 789.studio:~ *$ 
```

**Comment by EgonFreeman** [comment]
Issue fixed as of commit 0b59cef.
